pints/_transformation.py

```diff
@@ -940,6 +945,63 @@ def sample(self, n=1):
             self._transform.to_search(p) for p in self._boundaries.sample(n)]
 
 
+class TransformedRectangularBoundaries(pints.RectangularBoundaries):
+    """
+    A :class:`pints.RectangularBoundaries` that accepts parameters in a
+    transformed search space.
+
+    This transformation handles the special case where:
+
+    - the boundaries being transformed are :class:`pints.RectangularBoundaries`
+    - the transformation is element-wise.
+
+    When these conditions are met, the lower and upper boundaries can simply be
+    transformed and methods like :meth:`lower()` and :meth:`range()` can be
+    provided.
+
+    Not to be confused with :class:`pints.RectangularBoundariesTransformation`.
+
+    Extends :class:`pints.RectangularBoundaries`.
+
+    Parameters
+    ----------
+    boundaries
+        A :class:`pints.RectangularBoundaries` object.
+    transformation
+        An element-wise transformation.
+    """
+    def __init__(self, boundaries, transformation):
+
+        # Check input
+        if not isinstance(boundaries, pints.RectangularBoundaries):
+            raise ValueError('A TransformedRectangularBoundaries can only be'
+                             ' created from a RectangularBoundaries object.')
+        if not transformation.elementwise():
+            raise ValueError('A TransformedRectangularBoundaries can only be'
+                             ' created from an element-wise transformation.')
+        if transformation.n_parameters() != boundaries.n_parameters():
+            raise ValueError('Number of parameters for boundaries and '
+                             'transformation must match.')
+
+        # Transform upper and lower boundaries
+        a = transformation.to_search(boundaries.lower())
+        b = transformation.to_search(boundaries.upper())
+        lower = np.minimum(a, b)
+        upper = np.maximum(a, b)
+
+        # Pass transformed boundaries to RectangularBoundaries
+        super().__init__(lower, upper)
+
+        # Store input
+        self._bound = boundaries
+        self._trans = transformation
```

Comment:
It doesn't matter too much, but maybe use the same naming convention as above? So `self._boundaries` and `self._transform`?